projects/igniteui-angular/src/lib/services/exporter-common/base-export-service.ts

```diff
@@ -274,6 +275,9 @@ export abstract class IgxBaseExporter {
         this.prepareData(grid);
         this.addLevelData();
         this.addPivotGridColumns(grid);
+        if (grid.pivotUI.showRowHeaders) {
```

Comment:
I think would be better to move all that in the `addPivotGridColumns` method so everything is in one place.